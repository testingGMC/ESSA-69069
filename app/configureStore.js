/**
 * Create the store with dynamic reducers
 */

import { createStore, applyMiddleware, compose } from 'redux'
import { routerMiddleware } from 'connected-react-router'
import createSagaMiddleware from 'redux-saga'
import { NOTIFICATION_TYPES } from 'shared/constants'
import { pushNotification } from 'shared/utils/notification'
import { errorsExtraction } from 'shared/utils/error-interceptors'
import reduxMiddlewares from './reduxMiddlewares'
import createReducer from './reducers'

export default function configureStore(initialState = {}, history) {
  let composeEnhancers = compose
  const reduxSagaMonitorOptions = {}

  // If Redux Dev Tools and Saga Dev Tools Extensions are installed, enable them
  /* istanbul ignore next */
  if (process.env.NODE_ENV !== 'production' && typeof window === 'object') {
    /* eslint-disable no-underscore-dangle */
    if (window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__)
      composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({})

    // NOTE: Uncomment the code below to restore support for Redux Saga
    // Dev Tools once it supports redux-saga version 1.x.x
    // if (window.__SAGA_MONITOR_EXTENSION__)
    //   reduxSagaMonitorOptions = {
    //     sagaMonitor: window.__SAGA_MONITOR_EXTENSION__,
    //   };
    /* eslint-enable */
  }

  // const error400Handling = store => next => action => {
  //   if (action.e && action.type.slice(-7) === 'FAILURE') {
  //     if (action.e.response.status === 400) {
  //       const errors = errorsExtraction(action.e.response.data.model)
  //       errors.forEach(error =>
  //         pushNotification(NOTIFICATION_TYPES.error, error),
  //       )
  //       next({ ...action, errors })
  //       return
  //     }
  //   }
  //   next(action)
  // }

  const sagaMiddleware = createSagaMiddleware(reduxSagaMonitorOptions)

  // Create the store with two middlewares
  // 1. sagaMiddleware: Makes redux-sagas work
  // 2. routerMiddleware: Syncs the location/URL path to the state
  const middlewares = [
    sagaMiddleware,
    routerMiddleware(history),
    ...reduxMiddlewares,
  ]

  const enhancers = [applyMiddleware(...middlewares)]

  const store = createStore(
    createReducer(),
    initialState,
    composeEnhancers(...enhancers),
  )

  // Extensions
  store.runSaga = sagaMiddleware.run
  store.injectedReducers = {} // Reducer registry
  store.injectedSagas = {} // Saga registry

  // Make reducers hot reloadable, see http://mxs.is/googmo
  /* istanbul ignore next */
  if (module.hot) {
    module.hot.accept('./reducers', () => {
      store.replaceReducer(createReducer(store.injectedReducers))
    })
  }

  return store
}
