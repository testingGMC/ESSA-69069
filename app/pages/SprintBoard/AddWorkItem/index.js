import AddWorkItem from './AddWorkItem'

export default AddWorkItem
