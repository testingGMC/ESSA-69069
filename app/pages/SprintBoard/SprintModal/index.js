import { Form } from 'antd'
import SprintModal from './SprintModal'
export default Form.create({ name: 'manage_sprint' })(SprintModal)
