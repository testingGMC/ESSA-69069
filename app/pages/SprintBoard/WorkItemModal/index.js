import { Form } from 'antd'
import WorkItemModal from './WorkItemModal'
export default Form.create({ name: 'add_work_item' })(WorkItemModal)
