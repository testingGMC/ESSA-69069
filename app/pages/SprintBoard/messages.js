/*
 * SprintBoard Messages
 *
 * This contains all the text for the SprintBoard container.
 */

import { defineMessages } from 'react-intl'

export const scope = 'app.containers.SprintBoard'

export default defineMessages({
  helmetTitle: {
    id: `${scope}.helmetTitle`,
    defaultMessage: 'Sprint Board',
  },
  helmetDescription: {
    id: `${scope}.helmetDescription`,
    defaultMessage: '"Description of Sprint Board',
  },
  delete: {
    id: `${scope}.delete`,
    defaultMessage: 'Delete',
  },

  confirmMessage: {
    id: `${scope}.confirmMessage`,
    defaultMessage: 'Are you sure to delete this workitem?',
  },
  yes: {
    id: `${scope}.yes`,
    defaultMessage: 'Yes',
  },
  no: {
    id: `${scope}.no`,
    defaultMessage: 'No',
  },
  update: {
    id: `${scope}.update`,
    defaultMessage: 'Update',
  },
  createSprint: {
    id: `${scope}.addSprint`,
    defaultMessage: 'Create Sprint',
  },
  workDays: {
    id: `${scope}.workDays`,
    defaultMessage: 'Work Days',
  },
  todo: {
    id: `${scope}.todo`,
    defaultMessage: 'TO DO',
  },
  active: {
    id: `${scope}.active`,
    defaultMessage: 'ACTIVE',
  },
  resolved: {
    id: `${scope}.resolved`,
    defaultMessage: 'RESOLVED',
  },
  done: {
    id: `${scope}.done`,
    defaultMessage: 'DONE',
  },
  noSprintsTitle: {
    id: `${scope}.noSprints`,
    defaultMessage: 'No active sprint',
  },
  noSprintsContent: {
    id: `${scope}.noSprints`,
    defaultMessage: 'There is no active sprint. Add a new sprint',
  },
  chooseSprint: {
    id: `${scope}.chooseSprint`,
    defaultMessage: 'Choose sprint',
  },
})
