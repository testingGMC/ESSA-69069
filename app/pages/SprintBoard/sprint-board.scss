.sprint-board {
    margin-top: 10px;
    .drag-region {
        height: 100px;
    }
    .row-title {
        align-items: center;
        margin-bottom: 27px;
        background: #fff !important;
        padding: 0px 47px 0px 47px;
        margin-right: 0px !important;

        .ant-page-header {
            display: inline-flex;
            align-items: center;
            padding: 16px 4px;


            .ant-page-header-title-view-title {
                font-size: 28px;
            }
        }

        .sprint-filter {
            text-align: right;
           



        }

        .vertical-divider {
            height: 50px;
            margin: 0 30px;
        }

        .sprint-dates {
            color: #979797;
            font-size: 17px;
            font-weight: bold;


            .subtitle {
                font-weight: 600;
                color: #e0e0e0;
                font-size: 13px;
            }
        }
    }

    .content {
        padding: 27px 23px 47px 35px;
        max-width: 1200px;
        margin: 0 auto;

        .plus {
            margin-top: 20px;
            border: solid 1px #ebebeb;
            padding: 10px;
            text-align: center;
            background-color: #fff;
            color: #cecece;
            font-size: 18px;
            cursor: pointer;

        }



    }

    .title {
        color: #979797;
        font-size: 12px;
        font-weight: 800;
        margin-bottom: 30px;

    }

}
.ant-select-dropdown{
    .sprint-item {
        padding: 8px;
        cursor: pointer;
        color: #2a4bd8;
    }
}