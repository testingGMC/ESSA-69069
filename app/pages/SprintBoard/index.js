/**
 *
 * SprintBoard
 *
 */

import React, { memo } from 'react'
import { connect } from 'react-redux'

import { createStructuredSelector } from 'reselect'
import { compose } from 'redux'
import injectSaga from 'utils/injectSaga'
import { DAEMON } from 'utils/constants'
import { useInjectReducer } from 'utils/injectReducer'

import makeSelectProjectDetails, {
  getMembers,
} from 'shared/redux/selectedproject/selectors'
import reducerProjectDetails from 'shared/redux/selectedproject/reducer'
import * as actionsProjectDetails from 'shared/redux/selectedproject/actions'
import sagaProjectDetails from 'shared/redux/selectedproject/saga'

import makeSelectSprints, {
  getSelectedSprintId,
  getErrors,
} from 'shared/redux/sprints/selectors'
import reducer from 'shared/redux/sprints/reducer'
import * as actions from 'shared/redux/sprints/actions'
import saga from 'shared/redux/sprints/saga'
import { push } from 'connected-react-router'
import SprintBoard from './SprintBoard'

const SprintBoardIndex = props => {
  useInjectReducer({ key: 'sprints', reducer })
  useInjectReducer({ key: 'selectedProject', reducer: reducerProjectDetails })

  return <SprintBoard {...props} />
}

const mapStateToProps = createStructuredSelector({
  sprints: makeSelectSprints(),
  selectedProject: makeSelectProjectDetails(),
  selectedSprintId: getSelectedSprintId(),
  errors: getErrors(),
  members: getMembers(),
})

const mapDispatchToProps = {
  ...actions,
  ...actionsProjectDetails,
  push,
}

const withSaga = injectSaga({ key: 'sprints', saga, mode: DAEMON })
const withSagaProjectDetails = injectSaga({
  key: 'selectedProject',
  saga: sagaProjectDetails,
  mode: DAEMON,
})

const withConnect = connect(
  mapStateToProps,
  mapDispatchToProps,
)

export default compose(
  withConnect,
  memo,
  withSaga,
  withSagaProjectDetails,
)(SprintBoardIndex)
