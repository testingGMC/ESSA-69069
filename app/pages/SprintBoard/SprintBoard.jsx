/* eslint-disable prettier/prettier */
/* eslint-disable jsx-a11y/no-static-element-interactions */
/* eslint-disable jsx-a11y/click-events-have-key-events */
/* eslint-disable jsx-a11y/anchor-is-valid */
/**
 *
 * SprintBoard
 *
 */

import React, { useState, useEffect } from 'react'
import { DragDropContext , Draggable,  Droppable} from 'react-beautiful-dnd'
import PropTypes from 'prop-types'
import moment from 'moment'
import {
  Popconfirm,
  Menu,
  Badge,
  Row,
  Col,
  Select,
  PageHeader,
  Divider,
  Icon,
  Spin,
  Alert,
} from 'antd'
import QueueAnim from 'rc-queue-anim'
import {
  FormattedDate,
  injectIntl,
  FormattedMessage,
  intlShape,
} from 'react-intl'
import { Helmet } from 'react-helmet'
import { isEmpty } from 'lodash'
import {
  TODO_STATE,
  RESOLVED_STATE,
  ACTIVE_STATE,
  DONE_STATE,
  TASK_TYPE,
  BUG_TYPE,
  EMPTY_VALUE,
  YEAR_MONTH_DAY_DATE_FORMAT,
} from 'shared/constants'
import { isContentArchitect, getUserId } from 'shared/utils/access-token'

import routes from 'shared/routes'
import messages from './messages'
import WorkItem from './WorkItem'
import WorkItemModal from './WorkItemModal'
import SprintModal from './SprintModal'
import AddWorkItem from './AddWorkItem'

import './sprint-board.scss'

const PostDate = injectIntl(({ date, intl }) => (
  <span title={intl.formatDate(date)}>
    <FormattedDate value={date} year="numeric" month="long" day="2-digit" />
  </span>
))

const SprintBoard = ({
  selectedSprintId,
  sprints,
  selectedProject,
  fetchSprintsFirstTime,
  updateWorkItemState,
  deleteWorkItem,
  fetchWorkItems,
  addWorkItem,
  updateWorkItem,
  addSprint,
  match: {
    params: { projectId },
  },
  intl,
  roles,
  errors,
  push,
  members,
  setIsAddWorkItemModalVisible,
  setIsUpdateWorkItemModalVisible,
  setIsAddSprintModalVisible,
}) => {
  const {
    isAddWorkItemModalVisible,
    isUpdateWorkItemModalVisible,
    isAddSprintModalVisible,
  } = sprints
  const [todoList, setTodoList] = useState([])
  const [activeList, setActiveList] = useState([])
  const [resolvedList, setResolvedList] = useState([])
  const [doneList, setDoneList] = useState([])
  // const [isAddWorkItemModalVisible, setIsAddWorkItemModalVisible] = useState(
  //   false,
  // )
  // const [
  //   isUpdateWorkItemModalVisible,
  //   setIsUpdateWorkItemModalVisible,
  // ] = useState(false)
  // const [isAddSprintModalVisible, setIsAddSprintModalVisible] = useState(false)
  const [isMenuOpenProps, setIsMenuOpenProps] = useState(false)

  const [selectedWorkItem, setSelectedWorkItem] = useState({})

  const [formRef, setFormRef] = useState(null)
  const [selectedSprint, setSelectedSprint] = useState({})
  const COLUMNS = [
    { id: TODO_STATE.toString(), 
      title: <FormattedMessage {...messages.todo} />, list: todoList },
    {
      id: ACTIVE_STATE.toString(),
      title: <FormattedMessage {...messages.active} />,
      list: activeList,
    },
    {
      id: RESOLVED_STATE.toString(),
      title: <FormattedMessage {...messages.resolved} />,
      list: resolvedList,
    },
    { id: DONE_STATE.toString(), title: <FormattedMessage {...messages.done} />, list: doneList },
  ]

  useEffect(() => {
    fetchSprintsFirstTime(projectId)
  }, [])

  useEffect(() => {
    if (
      !isEmpty(members) &&
      !members.find(member => member.memberId.toString() === getUserId().toString())
    ) {
      push(routes.PROJECT_DETAILS.linkPath(projectId))
    }
  }, [members])

  const onDragEnd = (end) => {
    if(end.source === undefined || end.destination === undefined || end.source === null || end.destination === null || end.source.droppableId === undefined || end.destination.droppableId === undefined || end.source.droppableId === null || end.destination.droppableId === null)
      return
 
    const workItems = COLUMNS[end.source.droppableId].list.filter(x => x.id === end.draggableId)
    if (workItems.length > 0) {
      if(workItems[0].type.toString() === TASK_TYPE.toString() && end.destination.droppableId.toString() !== RESOLVED_STATE.toString())
        onUpdateWorkItemState(end.destination.droppableId, end.draggableId)
      if(workItems[0].type.toString() === BUG_TYPE.toString() && end.destination.droppableId.toString() !== DONE_STATE.toString())
        onUpdateWorkItemState(end.destination.droppableId, end.draggableId)
    }
  };
  useEffect(() => {
    setTodoList([])
    setActiveList([])
    setResolvedList([])
    setDoneList([])
    setSelectedSprint('')
    if (!isEmpty(selectedSprintId) && selectedSprintId !== undefined) {
      const pos = sprints.data.findIndex(s => s.id === selectedSprintId)

      if (sprints.data[pos].workItems) {
        setTodoList(
          sprints.data[pos].workItems.filter(w => w.state === TODO_STATE),
        )
        setActiveList(
          sprints.data[pos].workItems.filter(w => w.state === ACTIVE_STATE),
        )
        setResolvedList(
          sprints.data[pos].workItems.filter(w => w.state === RESOLVED_STATE),
        )
        setDoneList(
          sprints.data[pos].workItems.filter(w => w.state === DONE_STATE),
        )
      }
      setSelectedSprint(sprints.data[pos])
    }
  }, [sprints])
  const onUpdateWorkItemState = (state, workItemId) => {
    updateWorkItemState(selectedSprintId, workItemId, { state: Number(state) })
  }

  const myMenu = (workItemId, workItem) => (
    <Menu>
      <Menu.Item>
        <Popconfirm
          placement="topLeft"
          title={<FormattedMessage {...messages.confirmMessage} />}
          onConfirm={() => {
            deleteWorkItem(selectedSprintId, workItemId)
            setIsAddWorkItemModalVisible(false)
          }}
          okText={<FormattedMessage {...messages.yes} />}
          cancelText={<FormattedMessage {...messages.no} />}
        >
          <a href="#">{<FormattedMessage {...messages.delete} />}</a>
        </Popconfirm>
      </Menu.Item>
      <Menu.Item>
        <a
          href="#"
          onClick={() => {
            setSelectedWorkItem(workItem)
            setIsUpdateWorkItemModalVisible(true)
            setIsAddWorkItemModalVisible(false)
          }}
        >
          {<FormattedMessage {...messages.update} />}
        </a>
      </Menu.Item>
    </Menu>
  )
  const handleAddWorkItem = (name, type) => {
    const workItem = {
      name,
      type: Number(type),
      state: 0,
      sprintId: selectedSprintId,
      projectId,
    }
    addWorkItem(workItem)
    // setIsAddWorkItemModalVisible(false)
  }
  const handleUpdateWorkItem = () => {
    const { form } = formRef.props
    form.validateFields((err, values) => {
      if (err) {
        return
      }
      // form.resetFields()
      let { state } = selectedWorkItem
      if (
        values.type === BUG_TYPE.toString() &&
        selectedWorkItem.state === DONE_STATE
      ) {
        state = RESOLVED_STATE
      } else if (
        values.type === TASK_TYPE.toString() &&
        selectedWorkItem.state === RESOLVED_STATE
      ) {
        state = DONE_STATE
      }
      const workItem = {
        ...selectedWorkItem,
        ...values,
        type: Number(values.type),
        projectId,
      }
      updateWorkItem(selectedSprintId, selectedWorkItem.id, workItem)
      if (state !== selectedWorkItem.state) {
        updateWorkItemState(selectedSprintId, selectedWorkItem.id, {
          state,
        })
      }
      // handleUpdateWorkItemModalClose()
    })
  }
  const handleUpdateWorkItemModalClose = () => {
    setIsUpdateWorkItemModalVisible(false)
  }

  const handleAddSprint = () => {
    const { form } = formRef.props
    form.validateFields((err, values) => {
      if (err) {
        return
      }
      const Sprint = {
        startDate: values.startDate,
        dueDate: values.dueDate,
        projectId,
      }
      addSprint(Sprint)
    })
  }
  // useEffect(() => {
  //   if (isEmpty(errors) && isAddSprintModalVisible) {
  //     const { form } = formRef.props
  //     form.resetFields()
  //     handleAddSprintModalClose()
  //   }
  // }, [errors])

  const handleAddSprintModalClose = () => {
    setIsAddSprintModalVisible(false)
  }

  const saveFormRef = formRefParameter => {
    setFormRef(formRefParameter)
  }

  const renderMemberFullName = memberId => {
    if (selectedProject.data.members) {
      const filteredMember = selectedProject.data.members.find(
        member => member.memberId.toString() === memberId.toString(),
      )
      if (filteredMember) {
        return filteredMember.fullName
      }
      return EMPTY_VALUE
    }
    return EMPTY_VALUE
  }

  const canDragWorkItem = () => {
    const { startDate } = selectedSprint
    const now = moment().format(YEAR_MONTH_DAY_DATE_FORMAT)
    if (now >= moment(startDate).format(YEAR_MONTH_DAY_DATE_FORMAT)) {
      return true
    }
    return false
  }

  const helmetMessages = {
    helmetTitle: intl.formatMessage({
      ...messages.helmetTitle,
    }),
    helmetDescription: intl.formatMessage({
      ...messages.helmetDescription,
    }),
  }
  const renderSprintPeriod = (startDate, dueDate) => {
    const period = Number(moment(dueDate).diff(startDate, 'days'))
    return `${period + 1} `
  }
  return (
    <div className="sprint-board">
      <Helmet>
        <title>{helmetMessages.helmetTitle}</title>
        <meta name="description" content={helmetMessages.helmetDescription} />
      </Helmet>

      <Row gutter={24} type="flex" className="row-title">
        <Col xs={24} sm={24} md={8} lg={8} xl={8}>
          {selectedProject.data && !sprints.loading && (
            <PageHeader
              onBack={() => {
                window.history.back()
              }}
              title={`${selectedProject.data.name}`}
            />
          )}
        </Col>
        <Col xs={24} sm={24} md={16} lg={16} xl={16}>
          <div className="sprint-filter global-flex-horizontal-end">
            {sprints.data && !sprints.loading && (
              <Select
                style={{ width: 150 }}
                size="small"
                showSearch
                filterOption={(input, option) =>
                  option.props.children
                    .toLowerCase()
                    .indexOf(input.toLowerCase()) >= 0
                }
                optionFilterProp="children"
                placeholder={<FormattedMessage {...messages.chooseSprint} />}
                value={selectedSprintId || undefined}
                onChange={value => {
                  fetchWorkItems(value)
                  setIsAddWorkItemModalVisible(false)
                }}
                dropdownRender={menu => (
                  <div>
                    {menu}
                    {isContentArchitect(roles) && (
                      <>
                        <Divider style={{ margin: '4px 0' }} />

                        <div
                          onMouseDown={() => {
                            setIsAddSprintModalVisible(true)
                            setIsAddWorkItemModalVisible(false)
                          }}
                          className="sprint-item"
                        >
                          <Icon type="plus" />{' '}
                          <FormattedMessage {...messages.createSprint} />
                        </div>
                      </>
                    )}
                  </div>
                )}
              >
                {sprints.data.map(s => (
                  <Select.Option key={s.id} value={s.id}>
                    {s.name}
                  </Select.Option>
                ))}
              </Select>
            )}
            {selectedSprint.startDate && selectedSprint.dueDate && (
              <>
                <Divider type="vertical" className="vertical-divider" />

                <div className="sprint-dates">
                  <PostDate date={selectedSprint.startDate} /> -&nbsp;
                  <PostDate date={selectedSprint.dueDate} />
                  <div className="subtitle">
                    {renderSprintPeriod(
                      selectedSprint.startDate,
                      selectedSprint.dueDate,
                    )}
                    <FormattedMessage {...messages.workDays} />
                  </div>
                </div>
              </>
            )}
          </div>
        </Col>
      </Row>
      <Spin spinning={sprints.loading}>
        <div className="content">
          <Row gutter={24}>
            {sprints.data && !isEmpty(sprints.data) ? (
              <DragDropContext
                // onDragStart={onDragStart}
                onDragEnd={onDragEnd}
              >
                {COLUMNS.map(column => (
                  <Col xs={24} sm={12} md={6} lg={6} xl={6} key={column.id}>
                    <div className="title">
                      <span>{column.title} </span>
                      <Badge
                        count={column.list.length}
                        style={{
                          marginLeft: '10px',
                          backgroundColor: '#f9efc8',
                          color: '#ffcf20',
                          fontWeight: 'bold',
                          width: '31px',
                        }}
                      />
                    </div>
                    <Droppable
                      droppableId={column.id}
                      onUpdateWorkItemState={onUpdateWorkItemState}
                      isDragDisabled={false}
                    >
                      {(provided, snapshot) => (
                        <div
                          ref={provided.innerRef}
                          style={{ backgroundColor: snapshot.isDraggingOver ? '#f2f2f2' : '#fafafa' }}
                          {...provided.droppableProps}
                        >
                          <QueueAnim component="WorkItem" type={['right', 'left']} leaveReverse>

                            {column.list.map((workItem, index) => (
                              <Draggable
                                draggableId={workItem.id}
                                key={workItem.id}
                                type={workItem.type}
                                isDragDisabled={!canDragWorkItem()}
                                index={index}
                              >
                                {(provided, snapshot) => (
                                  <div
                                    ref={provided.innerRef}
                                    {...provided.draggableProps}
                                    {...provided.dragHandleProps}
                                  >
                                    <WorkItem
                                      name={workItem.name}
                                      completionDate={
                                        workItem.completionDate && (
                                          <PostDate date={workItem.completionDate} />
                                        )
                                      }
                                      type={workItem.type}
                                      fullName={renderMemberFullName(workItem.assignedTo)}
                                      workItem={workItem}
                                      menu={myMenu(workItem.id, workItem)}
                                      isMenuOpen={isMenuOpenProps}
                                      setIsMenuOpen={setIsMenuOpenProps}
                                      setIsUpdateWorkItemModalVisible={
                                        setIsUpdateWorkItemModalVisible
                                      }
                                      isUpdateWorkItemModalVisible={
                                        isUpdateWorkItemModalVisible
                                      }
                                      setSelectedWorkItem={setSelectedWorkItem}
                                      setIsAddWorkItemModalVisible={
                                        setIsAddWorkItemModalVisible
                                      }
                                    />
                                  </div>
                                )} 
                              </Draggable>
                            ))}
                          </QueueAnim>
                          <div className="drag-region">

                          </div>
                          {provided.placeholder}
                        </div>
                      )}
                    </Droppable>
                    {column.id === TODO_STATE.toString() && selectedSprintId && (
                      <>
                        {isAddWorkItemModalVisible && (
                          <AddWorkItem
                            addWorkItem={handleAddWorkItem}
                            onCancel={() => setIsAddWorkItemModalVisible(false)}
                          />
                        )}
                        {!isAddWorkItemModalVisible && (
                          <div
                            className="plus"
                            onClick={() => setIsAddWorkItemModalVisible(true)}
                          >
                            +
                          </div>
                        )}
                      </>
                    )}
            
                  </Col>
                ))
                }</DragDropContext>) : (
              <Alert
                message={<FormattedMessage {...messages.noSprintsTitle} />}
                description={
                  <FormattedMessage {...messages.noSprintsContent} />
                }
                type="info"
              />
            )}
         
          </Row>
    
        </div>
      </Spin>
      {isUpdateWorkItemModalVisible && (
        <WorkItemModal
          wrappedComponentRef={saveFormRef}
          isVisible={isUpdateWorkItemModalVisible}
          handleCancel={handleUpdateWorkItemModalClose}
          handleOk={handleUpdateWorkItem}
          name={selectedWorkItem.name}
          description={selectedWorkItem.description}
          type={selectedWorkItem.type}
          assignTo={selectedWorkItem.assignTo}
          sprints={sprints.data}
          sprintId={selectedSprintId}
          members={selectedProject.data.members}
          memberId={selectedWorkItem.assignedTo}
        />
      )}

      {isAddSprintModalVisible && (
        <SprintModal
          wrappedComponentRef={saveFormRef}
          isVisible={isAddSprintModalVisible}
          handleCancel={handleAddSprintModalClose}
          handleOk={handleAddSprint}
          project={selectedProject.data}
        />
      )}
    </div>
  )
}

SprintBoard.propTypes = {
  sprints: PropTypes.object.isRequired,

  selectedSprintId: PropTypes.string.isRequired,
  selectedProject: PropTypes.object.isRequired,
  fetchSprintsFirstTime: PropTypes.func.isRequired,
  updateWorkItemState: PropTypes.func.isRequired,
  deleteWorkItem: PropTypes.func.isRequired,
  fetchWorkItems: PropTypes.func.isRequired,
  addWorkItem: PropTypes.func.isRequired,
  updateWorkItem: PropTypes.func.isRequired,
  addSprint: PropTypes.func.isRequired,
  match: PropTypes.object.isRequired,
  intl: intlShape.isRequired,
  roles: PropTypes.array.isRequired,
  errors: PropTypes.array,
  push: PropTypes.func.isRequired,
  members: PropTypes.array.isRequired,
}

export default injectIntl(SprintBoard)
