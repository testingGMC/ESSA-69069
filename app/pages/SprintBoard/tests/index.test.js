
import React from 'react'
import { render } from 'react-testing-library'
import { IntlProvider } from 'react-intl'
import { Provider } from 'react-redux'
import { browserHistory } from 'react-router-dom'

import SprintBoardIndex from '../index'
import SprintBoard from '../SprintBoard'
import { DEFAULT_LOCALE } from '../../../i18n'
import configureStore from '../../../configureStore'

const initialProps = {
  sprints: {},
  selectedSprintId: '',
  selectedProject: {},
  fetchSprintsFirstTime: jest.fn(),
  updateWorkItemState: jest.fn(),
  deleteWorkItem: jest.fn(),
  fetchWorkItems: jest.fn(),
  addWorkItem: jest.fn(),
  fetchProjectById: jest.fn(),
  updateWorkItem: jest.fn(),
  addSprint: jest.fn(),
  match: {},
  match: {
    params: { projectId: '1' },
  },
  // sprints: { data: {} }
}

describe('<SprintBoardIndex />', () => {
  let store
  beforeAll(() => {
    store = configureStore({}, browserHistory)
  })

  it('Expect to not log in console', () => {
    const spy = jest.spyOn(global.console, 'log')
    render(
      <Provider store={store}>
        <IntlProvider locale={DEFAULT_LOCALE}>
          <SprintBoardIndex {...initialProps} />
        </IntlProvider>
      </Provider>,
    )
    expect(spy).not.toHaveBeenCalled()
  })

  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error')
    render(
      <Provider store={store}>
        <IntlProvider locale={DEFAULT_LOCALE}>
          <SprintBoardIndex {...initialProps} />
        </IntlProvider>
      </Provider>,
    )
    expect(spy).not.toHaveBeenCalled()
  })
})

describe('<SprintBoard />', () => {
  let store
  beforeAll(() => {
    store = configureStore({}, browserHistory)
  })

  it('should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <Provider store={store}>
        <IntlProvider locale={DEFAULT_LOCALE}>
          <SprintBoard {...initialProps} />
        </IntlProvider>
      </Provider>,
    )
    expect(firstChild).toMatchSnapshot()
  })

  it('should fetch the SprintBoard on mount if the sprintId exists', () => {
    render(
      <Provider store={store}>
        <IntlProvider locale={DEFAULT_LOCALE}>
          <SprintBoard {...initialProps} />
        </IntlProvider>
      </Provider>,
    )
    expect(initialProps.fetchSprintsFirstTime).toHaveBeenCalled()
  })
})