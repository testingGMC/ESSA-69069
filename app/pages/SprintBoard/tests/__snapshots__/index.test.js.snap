// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SprintBoard /> should render and match the snapshot 1`] = `
<div
  class="sprint-board"
>
  <div
    class="ant-row-flex row-title"
    style="margin-left: -12px; margin-right: -12px;"
  >
    <div
      class="ant-col ant-col-xs-24 ant-col-sm-24 ant-col-md-8 ant-col-lg-8 ant-col-xl-8"
      style="padding-left: 12px; padding-right: 12px;"
    />
    <div
      class="ant-col ant-col-xs-24 ant-col-sm-24 ant-col-md-16 ant-col-lg-16 ant-col-xl-16"
      style="padding-left: 12px; padding-right: 12px;"
    >
      <div
        class="sprint-filter global-flex-horizontal-end"
      />
    </div>
  </div>
  <div
    class="ant-spin-nested-loading"
  >
    <div>
      <div
        class="ant-spin ant-spin-spinning"
      >
        <span
          class="ant-spin-dot ant-spin-dot-spin"
        >
          <i
            class="ant-spin-dot-item"
          />
          <i
            class="ant-spin-dot-item"
          />
          <i
            class="ant-spin-dot-item"
          />
          <i
            class="ant-spin-dot-item"
          />
        </span>
      </div>
    </div>
    <div
      class="ant-spin-container ant-spin-blur"
    >
      <div
        class="content"
      >
        <div
          class="ant-row"
          style="margin-left: -12px; margin-right: -12px;"
        >
          <div
            class="ant-alert ant-alert-info ant-alert-with-description ant-alert-no-icon"
            data-show="true"
          >
            <span
              class="ant-alert-message"
            >
              <span>
                No active sprint
              </span>
            </span>
            <span
              class="ant-alert-description"
            >
              <span>
                There is no active sprint. Add a new sprint
              </span>
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
